servicetalk-opentracing-zipkin-publisher/src/main/java/io/servicetalk/opentracing/zipkin/publisher/reporter/HttpReporter.java

```diff
@@ -0,0 +1,252 @@
+import static io.servicetalk.concurrent.api.Completable.completed;
+import static io.servicetalk.concurrent.api.Processors.newCompletableProcessor;
+import static io.servicetalk.concurrent.api.Processors.newPublisherProcessorDropHeadOnOverflow;
+import static io.servicetalk.concurrent.api.SourceAdapters.fromSource;
+import static io.servicetalk.concurrent.api.SourceAdapters.toSource;
+import static io.servicetalk.concurrent.internal.FutureUtils.awaitTermination;
+import static io.servicetalk.http.api.CharSequences.newAsciiString;
+import static io.servicetalk.http.api.HttpHeaderNames.CONTENT_TYPE;
+import static io.servicetalk.http.api.HttpHeaderValues.APPLICATION_JSON;
+import static java.time.Duration.ofSeconds;
+import static java.util.Objects.requireNonNull;
+
+/**
+ * A {@link Span} {@link Reporter} that will publish to an HTTP endpoint with a configurable encoding {@link Codec}.
+ */
+public final class HttpReporter extends Component implements Reporter<Span>, AsyncCloseable {
+    private static final Logger LOGGER = LoggerFactory.getLogger(HttpReporter.class);
+    static final String V1_PATH = "/api/v1/spans";
+    static final String V2_PATH = "/api/v2/spans";
+    static final CharSequence THRIFT_CONTENT_TYPE = newAsciiString("application/x-thrift");
+    static final CharSequence PROTO_CONTENT_TYPE = newAsciiString("application/protobuf");
+
+    private final PublisherSource.Processor<Span, Span> buffer;
+    private final AsyncCloseable closeable;
+
+    private volatile boolean closeInitiated;
+
+    private HttpReporter(final Builder builder) {
+        final HttpClient client = builder.clientBuilder.build();
+        SpanBytesEncoder spanEncoder = builder.codec.spanBytesEncoder();
+        final String path;
+        final CharSequence contentType;
+        switch (builder.codec) {
+            case JSON_V1:
+                path = V1_PATH;
+                contentType = APPLICATION_JSON;
+                break;
+            case JSON_V2:
+                path = V2_PATH;
+                contentType = APPLICATION_JSON;
+                break;
+            case THRIFT:
+                path = V2_PATH;
+                contentType = THRIFT_CONTENT_TYPE;
+                break;
+            case PROTO3:
+                path = V2_PATH;
+                contentType = PROTO_CONTENT_TYPE;
+                break;
+            default:
+                throw new IllegalArgumentException("Unknown codec: " + builder.codec);
+        }
+        final BufferAllocator allocator = client.executionContext().bufferAllocator();
+        final Publisher<Buffer> spans;
+        if (builder.disableBatching) {
+            buffer = newPublisherProcessorDropHeadOnOverflow(builder.maxConcurrentReports);
+            spans = fromSource(buffer).map(span -> allocator.wrap(spanEncoder.encode(span)));
+        } else {
+            buffer = newPublisherProcessorDropHeadOnOverflow(builder.batchSizeHint * builder.maxConcurrentReports);
```

Comment:
Can you please left a comment why we need to multiply `batchSizeHint` and `maxConcurrentReports`?